x-pack/heartbeat/security_linux.go

```diff
@@ -0,0 +1,222 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"os/user"
+	"runtime"
+	"strconv"
+	"syscall"
+
+	"github.com/elastic/beats/v7/libbeat/common/seccomp"
+	"kernel.org/pub/linux/libs/security/libcap/cap"
+)
+
+func init() {
+	// Here we set a bunch of linux specific security stuff.
+	// In the context of a container, where users frequently run as root, we follow BEAT_SETUID_AS to setuid/gid
+	// and add capabilities to make this actually run as a regular user. This also helps node in synthetics, which
+	// does not want to run as root. It's also just generally more secure.
+	if localUserName := os.Getenv("BEAT_SETUID_AS"); localUserName != "" && syscall.Geteuid() == 0 {
+		err := changeUser(localUserName)
+		if err != nil {
+			panic(err)
+		}
+		err = setCapabilities()
+		if err != nil {
+			panic(err)
+		}
+	}
+
+	switch runtime.GOARCH {
+	case "amd64", "386":
```

Comment:
You should test on arm64. The default policy is a deny list and it denies `execve`.

Reply:
That is good to know, I'd like to do that as a follow up PR to keep the scope down, since that is probably also broken before this